src/main/java/org/killbill/billing/plugin/bridge/api/converter/ConverterHelper.java

```diff
@@ -96,7 +96,7 @@ public static PaymentPluginStatus toPluginStatus(final String transactionStatusI
         if (transactions != null && !transactions.isEmpty()) {
             result = transactions.stream().filter(paymentTransaction -> (kbTransactionExternalKey != null && paymentTransaction.getTransactionExternalKey().equals(kbTransactionExternalKey)))
                     //.peek(paymentTransaction -> System.err.println(String.format(".... transactionId ='%s'", paymentTransaction.getTransactionId())))
-                    .collect(Collectors.collectingAndThen(Collectors.toList(), paymentTransactions -> paymentTransactions.isEmpty() ? transactions.get(transactions.size() - 1) : paymentTransactions.get(0)));
+                    .collect(Collectors.collectingAndThen(Collectors.toList(), paymentTransactions -> paymentTransactions.isEmpty() ? transactions.get(transactions.size() - 1) : paymentTransactions.get(paymentTransactions.size() - 1)));
```

Comment:
I am wondering if an even more robust fix, would be to look at all transactions for this `kbTransactionExternalKey`, ordered from the last one to the first one, and extract the `SUCCESS` one if it exists or default to latest if it does not exist. That would ensure that if for some reason we have concurrent retried and one succeeded and one fail, we end up with correct state.